charts/posthog/templates/clickhouse_instance.yaml

```diff
@@ -54,6 +54,10 @@ spec:
   templates:
     podTemplates:
       - name: pod-template
+          {{- if .Values.clickhouse.podAnnotations }}
```

Comment:
Nit: I've moved the if outside one line up so that the `pod-template` doesn't change by default (when changing it will trigger a pod reprovision).